Modules/System/Email/src/Message/EmailMessageImpl.Codeunit.al

```diff
@@ -324,6 +358,36 @@ codeunit 8905 "Email Message Impl."
         end;
     end;
 
+    procedure AddRecipient(RecipientType: Enum "Email Recipient Type"; Recipient: Text)
+    var
+        EmailRecipientRecord: Record "Email Recipient";
+        UniqueRecipients: Dictionary of [Text, Text];
+        Recipients: List of [Text];
+        tmpRecipient: Text;
+    begin
+        if Recipient <> '' then
+            exit;
+        EmailRecipientRecord.SetRange("Email Message Id", Message.Id);
+        EmailRecipientRecord.SetRange("Email Recipient Type", RecipientType);
+
+        if not EmailRecipientRecord.IsEmpty() then
+            EmailRecipientRecord.DeleteAll();
```

Comment:
Why deleting the existing recipients? That shouldn't be okay.

Reply:
You're absolutely right. I made a mistake when copying from the SetRecipients procedure, I've resolved it